Lib/fontTools/merge.py

```diff
@@ -919,21 +988,63 @@ def __init__(self, options=None):
 		self.options = options
 
 	def merge(self, fontfiles):
-
-		mega = ttLib.TTFont()
+		# Take first input file sfntVersion
+		sfntVersion = ttLib.TTFont(fontfiles[0]).sfntVersion
+		mega = ttLib.TTFont(sfntVersion=sfntVersion)
 
 		#
 		# Settle on a mega glyph order.
 		#
 		fonts = [ttLib.TTFont(fontfile) for fontfile in fontfiles]
+
+		
+		# Check if all fonts have CFF
+		allOTFs = all(cff is not None for cff in [font.get('CFF ') for font in fonts])
+
+		cffTables = []
+		if allOTFs:
+			for font in fonts:
+				# Desubroutinize with Subsetter (like compreffor does)
```

Comment:
we could factor this out into a separate function `_desubroutinize(otf)` or similar

Reply:
doing it